bftengine/src/bftengine/FullNodeReplica.cpp

```diff
@@ -0,0 +1,295 @@
+// Concord
+//
+// Copyright (c) 2018, 2019 VMware, Inc. All Rights Reserved.
+//
+// This product is licensed to you under the Apache 2.0 license (the "License").  You may not use this product except in
+// compliance with the Apache 2.0 License.
+//
+// This product may include a number of subcomponents with separate copyright notices and license terms. Your use of
+// these subcomponents is subject to the terms and conditions of the sub-component's license, as noted in the LICENSE
+// file.
+
+#include <optional>
+#include <functional>
+#include <bitset>
+
+#include <bftengine/Replica.hpp>
+#include <messages/StateTransferMsg.hpp>
+#include "FullNodeReplica.hpp"
```

Comment:
A general comment about this new file:
Our code style should be underscored for parameters.
If you already edit legacy code where the camel case is there, you keep the original style.
but this is a new file, so you should have used lower case underscored parameter names in both hpp/cpp
Example:
A general comment about this new file:
Our code style should be underscored for parameters.
If you already edit legacy code where the camel case is there, you keep the original style.
but this is a new file, so you should have used lower case underscored parameter names.
Example:
`replicasLastKnownEpochVal`
becomes
`replicas_last_known_epoch_val`